src/components/StateMachineProperties.tsx

```diff
@@ -38,42 +39,28 @@ interface StateMachinePropertiesCallbackProps {
 };
 
 const mapStateToProps = (state: XCSpyState): StateMachinePropertiesProps => {
+    const id = state.stateMachineProperties.id;
+    const active = state.stateMachineProperties.active;
+    const componentProperties = state.components.componentProperties;
+    const currentComponent = state.components.currentComponent;
+    const stateMachine = state.stateMachineProperties.stateMachine;
+    const instances = (!active) ? null : componentProperties[currentComponent].stateMachineProperties[stateMachine];
+    const ids = (instances !== null) ? Object.keys(instances) : null;
+    const stateMachineRef = (() => {
```

Comment:
Avoid anonymous function. (useless in this case)
Just do the same as the ids property